weldx/util.py

```diff
@@ -1201,3 +1202,58 @@ def time_ref(self, value: pd.Timestamp):
                 self._obj.time.attrs["time_ref"] = value  # set new time_ref value
             else:
                 self._obj.time.attrs["time_ref"] = value
+
+
+def _compare(x, y):
+    if isinstance(x, np.ndarray):
+        if not isinstance(y, np.ndarray):
+            return False
+        return np.all(x == y)
+
+    elif isinstance(x, xr.DataArray):
+        if not isinstance(y, xr.DataArray):
+            return False
+        return x.identical(y)
+
+    else:
+        return x == y
```

Comment:
What do you think of something like this ?
```python
_compare_funcs = {
    (np.ndarray, pint.Quantity): lambda x, y: np.all(x == y) if (type(x) == type(y)) else False,
    (xr.DataArray, xr.Dataset): lambda x, y: x.identical(y) if (type(x) == type(y)) else False,
}


def _compare(x, y):
    for key, item in _compare_funcs.items():
        if type(x) in key:
            return item(x, y)
    return x == y
```

Reply:
That is pretty elegant :)

Reply:
nice one, even though I remember that we had some trouble with the xarray `identical` function. But we can sort it out when it gets a problem